### PR TITLE
Lps 147675

### PR DESCRIPTION
**Pull request by amosfong**

### CODE REVIEW

**Comment by Sharryxue** [comment]
Failures are not specific to the commits here. Unrelated failures.

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/OrganizationResourceImpl.java [hunk 1]
```diff
@@ -317,6 +347,49 @@ public Organization putOrganization(
...
+	public Organization putOrganizationByExternalReferenceCode(
+			String externalReferenceCode, Organization organization)
+		throws Exception {
+
+		com.liferay.portal.kernel.model.Organization
+			serviceBuilderOrganization =
+				_organizationLocalService.fetchOrganizationByReferenceCode(
+					contextCompany.getCompanyId(), externalReferenceCode);
+
+		long countryId = _getCountryId(organization);
```
Comment (drewbrokke) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/OrganizationResourceImpl.java:360]: I think we want to null check this org before passing it into `_getCountryId`. Would you be willing to double-check that?

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/OrganizationResourceImpl.java [hunk 2]
```diff
@@ -317,6 +347,49 @@ public Organization putOrganization(
...
+			_organizationService.addOrUpdateOrganization(
+				externalReferenceCode,
+				_getDefaultParentOrganizationId(organization),
+				organization.getName(), type,
+				_getRegionId(organization, countryId), countryId, statusId,
+				organization.getComment(), false, null, site,
+				_getAddresses(organization), _getEmailAddresses(organization),
+				_getOrgLabors(organization), _getPhones(organization),
+				_getWebsites(organization),
+				ServiceContextFactory.getInstance(contextHttpServletRequest));
```
Comment (drewbrokke) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/OrganizationResourceImpl.java:385]: Would it work to just call `_organizationService.addOrUpdateOrganization` straight away? Or is is retrieving first needed for `type, `statusId`, `site`, etc?

**Review by drewbrokke — COMMENTED** [review]
Just a couple of questions on this one, @amosfong. Overall it looks good.

**Comment by drewbrokke** [comment]
ci:reevaluate:1337401_1274

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/OrganizationResourceImpl.java (hunk 1 above)
Comment (amosfong) [reply modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/OrganizationResourceImpl.java:360]: Hey Drew, I don't think we need a null check here. I think vulcan never passes a null value in the parameters? There are no null checks in any other methods.

Or did you confuse it with the serviceBuilderOrganization? Maybe the fetchOrganizationByReferenceCode()  can be moved down to right before it's used.

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/OrganizationResourceImpl.java (hunk 2 above)
Comment (amosfong) [reply modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/OrganizationResourceImpl.java:385]: Yea, those fields aren't updateable with the API, so I'm always getting the current value to pass to the addOrUpdate

**Comment by drewbrokke** [comment]
ci:test:relevant

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/OrganizationResourceImpl.java (hunk 1 above)
Comment (drewbrokke) [reply modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/OrganizationResourceImpl.java:360]: Ah, you're right I confused it with `serviceBuilderOrganization`. Yeah let's do an 'as used' SF edit here. The `countryId` isn't needed until right before the call to `_organizationService.addOrUpdateOrganization`, but I think the `_organizationLocalService.fetchOrganizationByReferenceCode` call can stay put. Thanks!

**Comment by amosfong** [comment]
https://github.com/liferay-usm/liferay-portal/pull/483

**Pull request closed without merging**